java/client/src/org/openqa/selenium/chrome/ChromeOptions.java

```diff
@@ -89,9 +76,78 @@ public ChromeOptions() {
   @Override
   public ChromeOptions merge(Capabilities extraCapabilities) {
     super.merge(extraCapabilities);
+    if (extraCapabilities instanceof ChromeOptions) {
+      Map options = (Map) extraCapabilities.asMap().get("goog:chromeOptions");
+      mergeArguments(options);
+      mergeExtensions(options);
+      mergeExtensionFiles(extraCapabilities);
+      mergeBinary(options);
+      mergeExperimentalOptions(extraCapabilities);
+    }
     return this;
   }
 
+  private void mergeArguments(Map options) {
+    addArguments((List<String>) options.get("args"));
+    deleteDuplicates((ArrayList) args);
```

Comment:
This `deleteDuplicates` phase would be avoided entirely if you just added all the arguments to an `ImmutableSet` or `LinkedHashSet`. These preserve insertion order, but won't allow identical instances.

Reply:
Alternatively, stream the new args, and only add them if they're not present.